src/device/wg_controller.rs

```diff
@@ -200,12 +199,60 @@ async fn consolidate_wg_peers<
     for key in delete_keys {
         telio_log_info!("Removing peer: {:?}", actual_peers.get(key));
         wireguard_interface.del_peer(*key).await?;
+
+        // Remove from session_keeper
+        if let Some(sk) = session_keeper {
+            sk.remove_node(key).await?;
+        }
     }
 
+    let batcher_threshold = features
+        .batching
+        .as_ref()
+        .map(|b| Duration::from_secs(b.direct_connection_threshold as u64));
+
     for key in insert_keys {
         telio_log_info!("Inserting peer: {:?}", requested_peers.get(key));
-        let peer = requested_peers.get(key).ok_or(Error::PeerNotFound)?;
-        wireguard_interface.add_peer(peer.peer.clone()).await?;
+        if features.batching.is_some() {
+            // If batching is enabled, we use the session keeper for the persistent keepalives
+            let mut peer = requested_peers
+                .get(key)
+                .ok_or(Error::PeerNotFound)?
+                .peer
+                .clone();
+
+            // Persistent keepalive interval has been set by the build_requested_peers_list for
+            // every node. It considers type of connection proxy/direct and special nodes as VPN or STUN.
+            // We want to disable WG keepalives and use our session keeper instead
+            let keepalive_interval = peer
+                .persistent_keepalive_interval
+                .map_or(Duration::from_secs(25), |k| Duration::from_secs(k as u64));
+            peer.persistent_keepalive_interval = Some(0); // Disable WG persistent keepalive
+            let ip_addresses = peer.ip_addresses.clone();
+
+            wireguard_interface.add_peer(peer).await?;
+
+            // Start keepalives for the new inserted node
+            if let Some(sk) = session_keeper {
+                let target = build_target(&ip_addresses, features.ipv6);
+                if target.0.is_some() || target.1.is_some() {
+                    sk.add_node(*key, target, keepalive_interval, batcher_threshold)
+                        .await?;
+                } else {
+                    telio_log_warn!("Peer {:?} has no ip address", key);
+                }
+            }
+        } else {
+            wireguard_interface
```

Comment:
Nit: It would be nice to have `wireguard_interface.add_peer()` without a conditional. Without `add_peer()` there's no connection and it only changes the `persistent-keepalive` based on batching. This can be refactored to be a bit nicer to read imho:
```
if features.batching.is_some() {
  session_keeper.add(peer);
  peer.peer.persistent_keepalive = Some(0);
}

wireguard_interface.add_peer(peer).await?;
```

This way it's less conditionals and no risk of `add_peer` not being executed due to being hidden inside of a condition